test/http-client-tests/src/main/java/software/amazon/awssdk/http/server/MockServer.java

```diff
@@ -150,10 +156,12 @@ public static class HalfCloseServerBehavior implements ServerBehaviorStrategy {
 
         @Override
         public void runServer(ServerSocket serverSocket) {
+            Socket socket = null;
             try {
                 while (true) {
-                    Socket socket = null;
+
                     try {
+                        Thread.sleep(200);
```

Comment:
Is there any reason we need to add a sleep?

Reply:
Yes, since we are interrupting when we stopServer(), we include a sleep to ensure the interrupt signal is received. Otherwise, the accept() call continues waiting until the join timeout is over.

Reply:
I removed the sleep . Now we will return if socket exception are seen in case of halfclose and preclose.
While for success case we wait until stopserver interrupts the listener thread,